### ISSUE TITLE
TypeError when father class defined __init_subclass__

### ISSUE THREAD

**Issue by xbanke** [seandstewart/typical#60]
* typical version: 2.0.3
* Python version: 3.7.4
* Operating System:

### Description
As the title said. If I subclass a base class with `__init_subclass__`, the method `super` will raise `TypeError`. 

```py
import typic


@typic.klass
class Base:
    def __init_subclass__(cls, **kwargs):
        print('subclassed!')
        super().__init_subclass__(**kwargs)


@typic.klass
class Foo(Base):
    foo: str
```

I guess this might be related to #58

**Comment by seandstewart** [comment]
Looks like in cleaning up the MRO for these class definitions, I broke Python's `super` magic. Fix incoming.

**Comment by seandstewart** [comment]
This is now patched in v2.0.4.

**Comment by xbanke** [comment]
Good job.
